var/spack/repos/builtin/packages/py-griddataformats/package.py

```diff
@@ -19,10 +19,13 @@ class PyGriddataformats(PythonPackage):
 
     license("LGPL-3.0-only")
 
+    version("1.0.2", sha256="b93cf7f36fce33dbc428026f26dba560d5c7ba2387caca495bad920f90094502")
     version("1.0.1", sha256="ad2c9ab7d672a6d8c426de7d083eee4f3e2b0bd59391675d30683c768ab83cc4")
 
     depends_on("py-setuptools", type="build")
-    depends_on("python@3.8:", type=("build", "run"))
-    depends_on("py-numpy@1.19:", type=("build", "run"))
+    depends_on("python@3.8:3.11", when="@1.0.1", type=("build", "run"))
+    depends_on("python@3.9:3.12", when="@1.0.2", type=("build", "run"))
```

Comment:
Might be better as open ended when as insurance against inattentitve updaters:
```suggestion
    depends_on("python@3.9:3.12", when="@1.0.2:", type=("build", "run"))
```